### PR TITLE
[DPE-5484] Add TLS flag + TLS CA fields

### PR DESCRIPTION
**Pull request by lucasgameiroborges**
Adds a `tls` flag to indicate wether TLS is being used on the provider's side + `tls-ca` field for the CA chain of certificates.

See corresponding implementation PR here: https://github.com/canonical/postgresql-k8s-operator/pull/719

### CODE REVIEW

**Comment by tonyandrewmeyer** [comment]
Hi - is there background somewhere on why the postgresql_client interface has manually crafted JSON rather than a schema.py that auto-generates the JSON?

**Comment by lucasgameiroborges** [comment]
Hi there - Not sure who to ping to get a review here? @tonyandrewmeyer or maybe @PietroPasotti ?

**Comment by tonyandrewmeyer** [comment]
> Hi, @tonyandrewmeyer! I believe it's because the `postgresql_client` and the other Data Platform interfaces were created in either 2022 or early 2023. I see that maybe the [first PR](https://github.com/canonical/charm-relation-interfaces/pull/49) that used `schema.py` to auto-generate the JSON was introduced in March 2023.

Ok, thanks. I think we'll want to have all the interfaces defined the same way (manual schema.py, automated JSON) eventually, but I guess it doesn't have to be done now.

File: interfaces/postgresql_client/v0/README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ If any side, Provider or Requirer doesn't support Juju Secrets, sensitive inform
 - Is expected to provide the `uris` field with the connection string, in libpq's URI format, which can be used for direct connection to the db.
 - Is expected to provide optional `read-only-endpoints` field with a comma-separated list of hosts or one Kubernetes Service, which can be used for Read-only queries.
 - Is expected to provide the `version` field whenever database charm wants to communicate its database version.
+- Is expected to provide the `tls` field flag, indicating wether the provider has TLS enabled or not.
```
Comment (tonyandrewmeyer) [inline interfaces/postgresql_client/v0/README.md:38]:
```suggestion
- Is expected to provide the `tls` field flag, indicating whether the provider has TLS enabled or not.
```

**Review by tonyandrewmeyer — APPROVED** [review]
LGTM - one small typo.

File: interfaces/postgresql_client/v0/schemas/provider.json [hunk 2]
```diff
@@ -102,6 +102,27 @@
             "examples": [
                 "ready"
             ]
+        },
+        "tls": {
+            "$id": "#/properties/tls",
+            "title": "TLS",
+            "description": "Flag that indicates wether TLS is being used by the PostgreSQL charm or not",
```
Comment (tonyandrewmeyer) [inline interfaces/postgresql_client/v0/schemas/provider.json:109]: Nit:

```suggestion
            "description": "Flag that indicates whether TLS is being used by the PostgreSQL charm or not",
```

**Review by tonyandrewmeyer — APPROVED** [review]
LGTM, thanks!

**Pull request merged**